controllers/logicalvolume_controller.go

```diff
@@ -194,16 +195,51 @@ func (r *LogicalVolumeReconciler) createLV(ctx context.Context, log logr.Logger,
 			return nil
 		}
 
-		resp, err := r.lvService.CreateLV(ctx, &proto.CreateLVRequest{Name: string(lv.UID), DeviceClass: lv.Spec.DeviceClass, SizeGb: uint64(reqBytes >> 30)})
-		if err != nil {
-			code, message := extractFromError(err)
-			log.Error(err, message)
-			lv.Status.Code = code
-			lv.Status.Message = message
-			return err
+		var volume *proto.LogicalVolume
+
+		// Create a snapshot LV
+		if lv.Spec.Snapshot.DataSource != "" {
+			switch lv.Spec.Snapshot.AccessType {
+			case "ro", "rw":
+			default:
+				return fmt.Errorf("invalid access type for source volume: %s", lv.Spec.Snapshot.AccessType)
```

Comment:
I don't think access type will failback to "ro" as it is set:

- Explicitly as "to" for kubernetes snapshots
- and as "rw" for restored volumes

Let me know if I am missing anything

Reply:
Since the `accessType` is automatically set by the TopoLVM itself and not by any user, the chances of it setting to something unexpected are less.
And even if it is left empty, It is better to catch it then and there as an `invalid accessType error`; as if an lvm-snapshot is being created with an intent to be used as a PVC-Restore, and it's accessType is not set, falling back to `ro` will later fail with mounting failures, which will be hard to debug and trace back to accesType.

What do you think?